src/cart/crt_rpc.c

```diff
@@ -1689,6 +1689,12 @@ crt_rpc_common_hdlr(struct crt_rpc_priv *rpc_priv)
 					crt_ctx->cc_rpc_cb_arg);
 	} else {
 		rpc_priv->crp_opc_info->coi_rpc_cb(&rpc_priv->crp_pub);
+		/*
+		 * Correspond to crt_rpc_handler_common -> crt_rpc_priv_init's
+		 * set refcount as 1.
+		 */
+		if (rpc_priv->crp_srv)
+			RPC_DECREF(rpc_priv);
```

Comment:
since you removed crt_rpc_cb_customized() calling above (in crt_hg.c, crt_rpc_handler_common), why not just remove this code branch of "else" (L1691 ~ L1698)?
now in DAOS usage all goes to "if (crt_rpc_cb_customized(crt_ctx, &rpc_priv->crp_pub)) ()" branch.

Reply:
Not all. Few CaRT tests still use other path. Tests fails if I remove it.